From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20=C4=8Cerm=C3=A1k?= <sairon@sairon.cz>
Date: Mon, 17 Aug 2026 14:48:21 +0200
Subject: [PATCH] ARM: dts: bcm2712: Scan sdio1 only once at boot
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Yellow has no SD card slot and without the non-removable property it
keeps polling the onboard eMMC on CM5 Lite forever. Fix it by indicating
it's non-removable, polling the interface only once at boot.

Signed-off-by: Jan Čermák <sairon@sairon.cz>
---
 arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5-ha-yellow.dts | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5-ha-yellow.dts b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5-ha-yellow.dts
index 0dbb19898636..b85fa4c0b777 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5-ha-yellow.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5-ha-yellow.dts
@@ -97,6 +97,11 @@ &uart10 {
 	skip-init;
 };
 
+/* Scan the eMMC only once at boot, Yellow has no microSD slot. */
+&sdio1 {
+	non-removable;
+};
+
 /* RP1 USB ports are not connected on Yellow. */
 &rp1_usb0 {
 	status = "disabled";
